nearswap/src/pool.rs

```diff
@@ -41,6 +42,16 @@ pub struct Pool {
     pub shares: LookupMap<AccountId, Balance>,
     /// check `PoolInfo.total_shares`
     pub total_shares: Balance,
+
+    populated: usize,
+    last_updated_index: usize,
+    pivoted: bool,
+    pub observation: Vec<Twap>,
+
+    pub mean_1min: (U128, U128),
+    pub mean_5min: (U128, U128),
+    pub mean_1h: (U128, U128),
+    pub mean_12h: (U128, U128)
```

Comment:
how about encapsulating all TWAP related things in a separate structure? 
1. Rename Twap -> Observatoin
2. put all twap related things in a new structure: Twap
```
pub struct Pool {
   ...
   twap Twap
}
```

Reply:
yes, thanks for suggestion